File: python/1160.py
def anos_para_ultrapassar(pa, pb, g1, g2):
    anos = 0
    while pa <= pb:
        pa += int(pa * g1 / 100)
        pb += int(pb * g2 / 100)
        anos += 1
        if anos > 100:
            return "Mais de 1 seculo."
    return f"{anos} anos."

t = int(input())

list = []

for _ in range(t):
    pa, pb, g1, g2 = map(float, input().split())
    resultado = anos_para_ultrapassar(int(pa), int(pb), g1, g2)
    list.append(resultado)

for response in list:
    print(response)

File: javascript/1160.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');

function populationIncrease(population, growingRate) {
    var growing = Number.parseInt(population * (growingRate / 100));

    if (growing > 0) {
        return growing;
    }

    return 0;
}

function calculate(populationA, populationB, growingRateA, growingRateB) {
    var futurePopulationA = populationA;
    var futurePopulationB = populationB;
    var year = 0;

    while (futurePopulationA <= futurePopulationB) {
        year++;

        if (year > 100) {
            break;
        }

        futurePopulationA += populationIncrease(
            futurePopulationA,
            growingRateA
        );
        futurePopulationB += populationIncrease(
            futurePopulationB,
            growingRateB
        );
    }

    return year;
}

function main() {
    var lines = input.split("\n");
    var quantity = Number(lines.shift());

    for (var i = 0; i < quantity; i++) {
        var data = lines[i].split(" ").map((value) => Number(value));
        var year = calculate(...data);

        if (year > 100) {
            process.stdout.write(`Mais de 1 seculo.\n`);
        } else {
            process.stdout.write(`${year} anos.\n`);
        }
    }
}

main();